src/Veldrid/GraphicsDevice.cs

```diff
@@ -718,6 +718,10 @@ public void Unmap(MappableResource resource, uint subresource)
                 throw new VeldridException(
                     $"The data size given to UpdateBuffer is too large. The given buffer can only hold {buffer.SizeInBytes} total bytes. The requested update would require {bufferOffsetInBytes + sizeInBytes} bytes.");
             }
+            if (sizeInBytes == 0)
+            {
+                return;
```

Comment:
If one graphics device already fails (vulkan) when performing an "update buffer" call with zero bytes, then this should probably throw a `VeldridException` instead of just silently returning. Especially since this is at the surface of the API.

Reply:
I would disagree here. This check is already implemented for the other rather similar APIs like `CommandList.UpdateBuffer`, `CopyBuffer` etc. and was discussed in #117 

For the user ignoring the call for `sizeInBytes==0` is not an error, all data supposed to be updated *was* updated, just that all data is no data in this case.

Reply:
I agree with it not throwing an exception. I don't think a device failing is a good reason for _all_ devices to fail, as Veldrid is itself opinionated on how operations are performed so that they conform to the requirements set by the device (the D3D buffer update process as an example).